Build of isajson-ena on main branch failed

I just tried to build a new ENA converter from the `repository-services/isajson-ena` and it failed.
To reproduce the failed build, we could:

`git clone https://github.com/elixir-europe/MARS test-mars`
`cd test-mars/repository-services/isajson-ena`
`./gradlew build -x test`
Error message is: 

```
> Task :spotlessJavaCheck FAILED



FAILURE: Build failed with an exception.



* What went wrong:

Execution failed for task ':spotlessJavaCheck'.

> The following files had format violations:

      src/main/java/com/elixir/biohackaton/ISAToSRA/controller/WebinIsaToXmlSubmissionController.java

          @@ -8,9 +8,9 @@

           import·com.elixir.biohackaton.ISAToSRA.model.Investigation;

           import·com.elixir.biohackaton.ISAToSRA.model.IsaJson;

           import·com.elixir.biohackaton.ISAToSRA.model.Study;

          -import·com.elixir.biohackaton.ISAToSRA.sra.service.*;

           import·com.elixir.biohackaton.ISAToSRA.sra.model.MarsReceipt;

           import·com.elixir.biohackaton.ISAToSRA.sra.model.Receipt;

          +import·com.elixir.biohackaton.ISAToSRA.sra.service.*;

           import·com.fasterxml.jackson.databind.ObjectMapper;

           import·io.swagger.v3.oas.annotations.responses.ApiResponse;

           import·io.swagger.v3.oas.annotations.responses.ApiResponses;

          @@ -32,42 +32,36 @@

           @Slf4j

           @RestController

           public·class·WebinIsaToXmlSubmissionController·{

          -··@Autowired

          -··private·BioSamplesAccessionsParser·bioSamplesAccessionsParser;

          +··@Autowired·private·BioSamplesAccessionsParser·bioSamplesAccessionsParser;

           

          -··@Autowired

          -··private·WebinStudyXmlCreator·webinStudyXmlCreator;

          +··@Autowired·private·WebinStudyXmlCreator·webinStudyXmlCreator;

           

          -··@Autowired

          -··private·WebinExperimentXmlCreator·webinExperimentXmlCreator;

          +··@Autowired·private·WebinExperimentXmlCreator·webinExperimentXmlCreator;

           

          -··@Autowired

          -··private·WebinProjectXmlCreator·webinProjectXmlCreator;

          +··@Autowired·private·WebinProjectXmlCreator·webinProjectXmlCreator;

           

          -··@Autowired

          -··private·WebinRunXmlCreator·webinRunXmlCreator;

          +··@Autowired·private·WebinRunXmlCreator·webinRunXmlCreator;

           

          -··@Autowired

          -··private·WebinHttpSubmissionService·webinHttpSubmissionService;

          +··@Autowired·private·WebinHttpSubmissionService·webinHttpSubmissionService;

           

          -··@Autowired

          -··private·ObjectMapper·objectMapper;

          +··@Autowired·private·ObjectMapper·objectMapper;

           

          -··@Autowired

          -··private·ReceiptConversionService·receiptConversionService;

          +··@Autowired·private·ReceiptConversionService·receiptConversionService;

           

          -··@Autowired

          -··private·ReceiptMarsService·receiptMarsService;

      ... (62 more lines that didn't fit)

  Violations also present in 10 other files.

  Run './gradlew :spotlessApply' to fix these violations.



* Try:

> Run with --stacktrace option to get the stack trace.

> Run with --info or --debug option to get more log output.

> Run with --scan to get full insights.



* Get more help at https://help.gradle.org



BUILD FAILED in 21s

7 actionable tasks: 6 executed, 1 up-to-date
```


Maybe there is a new way to build, which can be successful.
It will be great is someone can help.

--- a/.github/workflows/build-biosamples-service.yml
+++ b/.github/workflows/build-biosamples-service.yml
@@ -1,0 +1,32 @@
+name: Build and Push Docker image for BioSamples service
+
+on:
+  push:
+    paths:
+        - "repository-services/isajson-biosamples/**"
+  pull_request:
+    paths:
+        - "repository-services/isajson-biosamples/**"
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v6
+        with:
+          context: repository-services/isajson-biosamples
+          push: false
+          tags: user/app:latest

--- a/.github/workflows/build-ena-service.yml
+++ b/.github/workflows/build-ena-service.yml
@@ -1,0 +1,32 @@
+name: Build and Push Docker image for ENA service
+
+on:
+  push:
+    paths:
+        - "repository-services/isajson-ena/**"
+  pull_request:
+    paths:
+        - "repository-services/isajson-ena/**"
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v6
+        with:
+          context: repository-services/isajson-ena
+          push: false
+          tags: user/app:latest

--- a/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/controller/WebinIsaToXmlSubmissionController.java
+++ b/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/controller/WebinIsaToXmlSubmissionController.java
@@ -8,9 +8,9 @@ import com.elixir.biohackaton.ISAToSRA.biosamples.service.BioSamplesAccessionsPa
 import com.elixir.biohackaton.ISAToSRA.model.Investigation;
 import com.elixir.biohackaton.ISAToSRA.model.IsaJson;
 import com.elixir.biohackaton.ISAToSRA.model.Study;
-import com.elixir.biohackaton.ISAToSRA.sra.service.*;
 import com.elixir.biohackaton.ISAToSRA.sra.model.MarsReceipt;
 import com.elixir.biohackaton.ISAToSRA.sra.model.Receipt;
+import com.elixir.biohackaton.ISAToSRA.sra.service.*;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import io.swagger.v3.oas.annotations.responses.ApiResponse;
 import io.swagger.v3.oas.annotations.responses.ApiResponses;
@@ -32,42 +32,36 @@ import org.springframework.web.bind.annotation.RestController;
 @Slf4j
 @RestController
 public class WebinIsaToXmlSubmissionController {
-  @Autowired
-  private BioSamplesAccessionsParser bioSamplesAccessionsParser;
+  @Autowired private BioSamplesAccessionsParser bioSamplesAccessionsParser;
 
-  @Autowired
-  private WebinStudyXmlCreator webinStudyXmlCreator;
+  @Autowired private WebinStudyXmlCreator webinStudyXmlCreator;
 
-  @Autowired
-  private WebinExperimentXmlCreator webinExperimentXmlCreator;
+  @Autowired private WebinExperimentXmlCreator webinExperimentXmlCreator;
 
-  @Autowired
-  private WebinProjectXmlCreator webinProjectXmlCreator;
+  @Autowired private WebinProjectXmlCreator webinProjectXmlCreator;
 
-  @Autowired
-  private WebinRunXmlCreator webinRunXmlCreator;
+  @Autowired private WebinRunXmlCreator webinRunXmlCreator;
 
-  @Autowired
-  private WebinHttpSubmissionService webinHttpSubmissionService;
+  @Autowired private WebinHttpSubmissionService webinHttpSubmissionService;
 
-  @Autowired
-  private ObjectMapper objectMapper;
+  @Autowired private ObjectMapper objectMapper;
 
-  @Autowired
-  private ReceiptConversionService receiptConversionService;
+  @Autowired private ReceiptConversionService receiptConversionService;
 
-  @Autowired
-  private ReceiptMarsService receiptMarsService;
+  @Autowired private ReceiptMarsService receiptMarsService;
 
-  @ApiResponses(value = {
-      @ApiResponse(responseCode = "200", description = "Ok"),
-      @ApiResponse(responseCode = "401", description = "Unauthorized"),
-      @ApiResponse(responseCode = "403", description = "Forbidden"),
-      @ApiResponse(responseCode = "400", description = "Bad request"),
-      @ApiResponse(responseCode = "408", description = "Request Timeout"),
-      @ApiResponse(responseCode = "415", description = "Unsupported media type")
-  })
-  @PostMapping(value = "/submit", consumes = { APPLICATION_JSON_VALUE, APPLICATION_XML_VALUE })
+  @ApiResponses(
+      value = {
+        @ApiResponse(responseCode = "200", description = "Ok"),
+        @ApiResponse(responseCode = "401", description = "Unauthorized"),
+        @ApiResponse(responseCode = "403", description = "Forbidden"),
+        @ApiResponse(responseCode = "400", description = "Bad request"),
+        @ApiResponse(responseCode = "408", description = "Request Timeout"),
+        @ApiResponse(responseCode = "415", description = "Unsupported media type")
+      })
+  @PostMapping(
+      value = "/submit",
+      consumes = {APPLICATION_JSON_VALUE, APPLICATION_XML_VALUE})
   public String performSubmissionToEna(
       @RequestBody final String submissionPayload,
       @RequestParam(value = "webinUserName") String webinUserName,
@@ -83,8 +77,9 @@ public class WebinIsaToXmlSubmissionController {
 
     final IsaJson isaJson = this.objectMapper.readValue(submissionPayload, IsaJson.class);
     final List<Study> studies = getStudies(isaJson);
-    final Map<String, String> typeToBioSamplesAccessionMap = this.bioSamplesAccessionsParser
-        .parseIsaFileAndGetBioSampleAccessions(studies, new HashMap<>());
+    final Map<String, String> typeToBioSamplesAccessionMap =
+        this.bioSamplesAccessionsParser.parseIsaFileAndGetBioSampleAccessions(
+            studies, new HashMap<>());
 
     final Document document = DocumentHelper.createDocument();
     final Element webinElement = startPreparingWebinV2SubmissionXml(document);
@@ -93,8 +88,9 @@ public class WebinIsaToXmlSubmissionController {
     this.webinStudyXmlCreator.createENAStudySetElement(
         webinElement, studies, randomSubmissionIdentifier);
 
-    final Map<Integer, String> experimentSequenceMap = this.webinExperimentXmlCreator.createENAExperimentSetElement(
-        typeToBioSamplesAccessionMap, webinElement, studies, randomSubmissionIdentifier);
+    final Map<Integer, String> experimentSequenceMap =
+        this.webinExperimentXmlCreator.createENAExperimentSetElement(
+            typeToBioSamplesAccessionMap, webinElement, studies, randomSubmissionIdentifier);
 
     this.webinRunXmlCreator.createENARunSetElement(
         webinElement, studies, experimentSequenceMap, randomSubmissionIdentifier);
@@ -106,8 +102,9 @@ public class WebinIsaToXmlSubmissionController {
 
     writer.write(document);
 
-    final String receiptXml = webinHttpSubmissionService.performWebinSubmission(
-        webinUserName, document.asXML(), webinPassword);
+    final String receiptXml =
+        webinHttpSubmissionService.performWebinSubmission(
+            webinUserName, document.asXML(), webinPassword);
     final Receipt receiptJson = receiptConversionService.readReceiptXml(receiptXml);
     final MarsReceipt marsReceipt = receiptMarsService.convertReceiptToMars(receiptJson, isaJson);
 

--- a/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceipt.java
+++ b/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceipt.java
@@ -1,3 +1,4 @@
+/** Elixir BioHackathon 2022 */
 package com.elixir.biohackaton.ISAToSRA.sra.model;
 
 import lombok.Builder;
@@ -6,11 +7,11 @@ import lombok.Data;
 @Builder
 @Data
 public class MarsReceipt {
-    private String targetRepository;
+  private String targetRepository;
 
-    private MarsReceiptError[] errors;
+  private MarsReceiptError[] errors;
 
-    private MarsReceiptInfo[] info;
+  private MarsReceiptInfo[] info;
 
-    private MarsReceiptAccession[] accessions;
+  private MarsReceiptAccession[] accessions;
 }

--- a/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptAccession.java
+++ b/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptAccession.java
@@ -1,3 +1,4 @@
+/** Elixir BioHackathon 2022 */
 package com.elixir.biohackaton.ISAToSRA.sra.model;
 
 import lombok.Builder;
@@ -6,7 +7,7 @@ import lombok.Data;
 @Builder
 @Data
 public class MarsReceiptAccession {
-    private String value;
+  private String value;
 
-    private MarsReceiptPath[] path;
+  private MarsReceiptPath[] path;
 }

--- a/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptError.java
+++ b/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptError.java
@@ -1,18 +1,18 @@
+/** Elixir BioHackathon 2022 */
 package com.elixir.biohackaton.ISAToSRA.sra.model;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonInclude.Include;
-
 import lombok.Builder;
 import lombok.Data;
 
 @Builder
 @Data
 public class MarsReceiptError {
-    private MarsReceiptErrorType type;
+  private MarsReceiptErrorType type;
 
-    private String message;
+  private String message;
 
-    @JsonInclude(Include.NON_NULL)
-    private MarsReceiptPath[] path;
+  @JsonInclude(Include.NON_NULL)
+  private MarsReceiptPath[] path;
 }

--- a/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptErrorType.java
+++ b/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptErrorType.java
@@ -1,6 +1,7 @@
+/** Elixir BioHackathon 2022 */
 package com.elixir.biohackaton.ISAToSRA.sra.model;
 
 public enum MarsReceiptErrorType {
-    INVALID_METADATA,
-    INVALID_DATA;
+  INVALID_METADATA,
+  INVALID_DATA;
 }

--- a/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptInfo.java
+++ b/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptInfo.java
@@ -1,16 +1,16 @@
+/** Elixir BioHackathon 2022 */
 package com.elixir.biohackaton.ISAToSRA.sra.model;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonInclude.Include;
-
 import lombok.Builder;
 import lombok.Data;
 
 @Builder
 @Data
 public class MarsReceiptInfo {
-    @JsonInclude(Include.NON_NULL)
-    private String name;
+  @JsonInclude(Include.NON_NULL)
+  private String name;
 
-    private String message;
+  private String message;
 }

--- a/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptMessage.java
+++ b/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptMessage.java
@@ -1,18 +1,16 @@
+/** Elixir BioHackathon 2022 */
 package com.elixir.biohackaton.ISAToSRA.sra.model;
 
-import lombok.Builder;
-import lombok.Data;
-import lombok.Builder.Default;
-
-import java.util.List;
 import java.util.ArrayList;
+import java.util.List;
+import lombok.Builder;
+import lombok.Builder.Default;
+import lombok.Data;
 
 @Builder
 @Data
 public class MarsReceiptMessage {
-    @Default
-    private List<MarsReceiptError> errors = new ArrayList<>();
+  @Default private List<MarsReceiptError> errors = new ArrayList<>();
 
-    @Default
-    private List<MarsReceiptInfo> info= new ArrayList<>();
+  @Default private List<MarsReceiptInfo> info = new ArrayList<>();
 }

--- a/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptPath.java
+++ b/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptPath.java
@@ -1,16 +1,16 @@
+/** Elixir BioHackathon 2022 */
 package com.elixir.biohackaton.ISAToSRA.sra.model;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonInclude.Include;
-
 import lombok.Builder;
 import lombok.Data;
 
 @Builder
 @Data
 public class MarsReceiptPath {
-    private String key;
+  private String key;
 
-    @JsonInclude(Include.NON_NULL)
-    private MarsReceiptWhere where;
+  @JsonInclude(Include.NON_NULL)
+  private MarsReceiptWhere where;
 }

--- a/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptWhere.java
+++ b/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/model/MarsReceiptWhere.java
@@ -1,3 +1,4 @@
+/** Elixir BioHackathon 2022 */
 package com.elixir.biohackaton.ISAToSRA.sra.model;
 
 import lombok.Builder;
@@ -6,7 +7,7 @@ import lombok.Data;
 @Builder
 @Data
 public class MarsReceiptWhere {
-    private String key;
+  private String key;
 
-    private String value;
+  private String value;
 }

--- a/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/service/ReceiptMarsService.java
+++ b/repository-services/isajson-ena/src/main/java/com/elixir/biohackaton/ISAToSRA/sra/service/ReceiptMarsService.java
@@ -1,11 +1,5 @@
-/** Elixir BioHackathon 2023 */
+/** Elixir BioHackathon 2022 */
 package com.elixir.biohackaton.ISAToSRA.sra.service;
-
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Optional;
-
-import org.springframework.stereotype.Service;
 
 import com.elixir.biohackaton.ISAToSRA.model.DataFile;
 import com.elixir.biohackaton.ISAToSRA.model.IsaJson;
@@ -26,6 +20,10 @@ import com.elixir.biohackaton.ISAToSRA.sra.model.ReceiptObject;
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.SerializationFeature;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
 
 @Service
 public class ReceiptMarsService {
@@ -51,8 +49,9 @@ public class ReceiptMarsService {
 
   /**
    * Converting ENA receipt to Mars data format
-   * 
-   * @see https://github.com/elixir-europe/MARS/blob/refactor/repository-services/repository-api.md#response
+   *
+   * @see
+   *     https://github.com/elixir-europe/MARS/blob/refactor/repository-services/repository-api.md#response
    * @param receipt {@link Receipt} Receipt from ENA
    * @param isaJson {@link IsaJson} Requested ISA-Json
    * @return {@link MarsReceipt} Mars response data
@@ -71,42 +70,75 @@ public class ReceiptMarsService {
   }
 
   private void setMarsReceiptErrors(final Messages messages, final MarsReceiptMessage marsMessage) {
-    Optional.ofNullable(messages.getErrorMessages()).orElse(new ArrayList<>()).forEach(error -> {
-      marsMessage.getErrors().add(MarsReceiptError.builder()
-          .message(error)
-          .type(MarsReceiptErrorType.INVALID_METADATA)
-          .build());
-    });
+    Optional.ofNullable(messages.getErrorMessages())
+        .orElse(new ArrayList<>())
+        .forEach(
+            error -> {
+              marsMessage
+                  .getErrors()
+                  .add(
+                      MarsReceiptError.builder()
+                          .message(error)
+                          .type(MarsReceiptErrorType.INVALID_METADATA)
+                          .build());
+            });
   }
 
   private void setMarsReceiptInfo(final Messages messages, final MarsReceiptMessage marsMessage) {
-    Optional.ofNullable(messages.getInfoMessages()).orElse(new ArrayList<>()).forEach(info -> {
-      marsMessage.getInfo().add(MarsReceiptInfo.builder()
-          .message(info)
-          .build());
-    });
+    Optional.ofNullable(messages.getInfoMessages())
+        .orElse(new ArrayList<>())
+        .forEach(
+            info -> {
+              marsMessage.getInfo().add(MarsReceiptInfo.builder().message(info).build());
+            });
   }
 
   private MarsReceiptAccession[] getMarsAccessions(
-      final Receipt receipt,
-      final IsaJson isaJson,
-      final MarsReceiptMessage marsMessage) {
+      final Receipt receipt, final IsaJson isaJson, final MarsReceiptMessage marsMessage) {
     final List<MarsReceiptAccession> accessions = new ArrayList<>();
     if (receipt.isSuccess()) {
-      Optional.ofNullable(isaJson.investigation.studies).orElse(new ArrayList<>()).forEach(study -> {
-        setStudyMarsAccession(accessions, receipt, study, marsMessage);
-        Optional.ofNullable(study.materials.samples).orElse(new ArrayList<>()).forEach(sample -> {
-          setSampleMarsAccession(accessions, receipt, study.title, sample, marsMessage);
-        });
-        Optional.ofNullable(study.assays).orElse(new ArrayList<>()).forEach(assay -> {
-          Optional.ofNullable(assay.materials.otherMaterials).orElse(new ArrayList<>()).forEach(otherMaterial -> {
-            setExperimentMarsAccession(accessions, receipt, study.title, assay.id, otherMaterial, marsMessage);
-          });
-          Optional.ofNullable(assay.dataFiles).orElse(new ArrayList<>()).forEach(dataFile -> {
-            setRunMarsAccession(accessions, receipt, study.title, assay.id, dataFile, marsMessage);
-          });
-        });
-      });
+      Optional.ofNullable(isaJson.investigation.studies)
+          .orElse(new ArrayList<>())
+          .forEach(
+              study -> {
+                setStudyMarsAccession(accessions, receipt, study, marsMessage);
+                Optional.ofNullable(study.materials.samples)
+                    .orElse(new ArrayList<>())
+                    .forEach(
+                        sample -> {
+                          setSampleMarsAccession(
+                              accessions, receipt, study.title, sample, marsMessage);
+                        });
+                Optional.ofNullable(study.assays)
+                    .orElse(new ArrayList<>())
+                    .forEach(
+                        assay -> {
+                          Optional.ofNullable(assay.materials.otherMaterials)
+                              .orElse(new ArrayList<>())
+                              .forEach(
+                                  otherMaterial -> {
+                                    setExperimentMarsAccession(
+                                        accessions,
+                                        receipt,
+                                        study.title,
+                                        assay.id,
+                                        otherMaterial,
+                                        marsMessage);
+                                  });
+                          Optional.ofNullable(assay.dataFiles)
+                              .orElse(new ArrayList<>())
+                              .forEach(
+                                  dataFile -> {
+                                    setRunMarsAccession(
+                                        accessions,
+                                        receipt,
+                                        study.title,
+                                        assay.id,
+                                        dataFile,
+                                        marsMessage);
+                                  });
+                        });
+              });
     }
 
     return accessions.toArray(MarsReceiptAccession[]::new);
@@ -122,19 +154,24 @@ public class ReceiptMarsService {
       final Receipt receipt,
       final Study study,
       final MarsReceiptMessage marsMessage) {
-    final List<ReceiptObject> studies = Optional.ofNullable(receipt.getStudies()).orElse(receipt.getProjects());
-    final Optional<ReceiptObject> studyReceipt = Optional.ofNullable(studies)
-        .orElse(new ArrayList<>())
-        .stream()
-        .filter(s -> s.getAlias().equals(study.title))
-        .findAny();
+    final List<ReceiptObject> studies =
+        Optional.ofNullable(receipt.getStudies()).orElse(receipt.getProjects());
+    final Optional<ReceiptObject> studyReceipt =
+        Optional.ofNullable(studies).orElse(new ArrayList<>()).stream()
+            .filter(s -> s.getAlias().equals(study.title))
+            .findAny();
     final MarsReceiptAccession marsStudyReceipt = getStudyMarsAccession(study.title, studyReceipt);
     if (studyReceipt.isEmpty()) {
-      marsMessage.getErrors().add(MarsReceiptError.builder()
-          .message(String.format("Cannot find a Study with the alias %s in the ENA receipt", study.title))
-          .type(MarsReceiptErrorType.INVALID_METADATA)
-          .path(marsStudyReceipt.getPath())
-          .build());
+      marsMessage
+          .getErrors()
+          .add(
+              MarsReceiptError.builder()
+                  .message(
+                      String.format(
+                          "Cannot find a Study with the alias %s in the ENA receipt", study.title))
+                  .type(MarsReceiptErrorType.INVALID_METADATA)
+                  .path(marsStudyReceipt.getPath())
+                  .build());
       return;
     }
     accessions.add(marsStudyReceipt);
@@ -146,18 +183,23 @@ public class ReceiptMarsService {
       final String studyTitle,
       final Sample sample,
       final MarsReceiptMessage marsMessage) {
-    final Optional<ReceiptObject> sampleReceipt = Optional.ofNullable(receipt.getSamples())
-        .orElse(new ArrayList<>())
-        .stream()
-        .filter(s -> s.getAlias().equals(sample.id))
-        .findAny();
-    final MarsReceiptAccession marsSampleReceipt = getSampleMarsAccession(studyTitle, sample.id, sampleReceipt);
+    final Optional<ReceiptObject> sampleReceipt =
+        Optional.ofNullable(receipt.getSamples()).orElse(new ArrayList<>()).stream()
+            .filter(s -> s.getAlias().equals(sample.id))
+            .findAny();
+    final MarsReceiptAccession marsSampleReceipt =
+        getSampleMarsAccession(studyTitle, sample.id, sampleReceipt);
     if (sampleReceipt.isEmpty()) {
-      marsMessage.getErrors().add(MarsReceiptError.builder()
-          .message(String.format("Cannot find a Sample with the alias %s in the ENA receipt", sample.id))
-          .type(MarsReceiptErrorType.INVALID_METADATA)
-          .path(marsSampleReceipt.getPath())
-          .build());
+      marsMessage
+          .getErrors()
+          .add(
+              MarsReceiptError.builder()
+                  .message(
+                      String.format(
+                          "Cannot find a Sample with the alias %s in the ENA receipt", sample.id))
+                  .type(MarsReceiptErrorType.INVALID_METADATA)
+                  .path(marsSampleReceipt.getPath())
+                  .build());
       return;
     }
     accessions.add(marsSampleReceipt);
@@ -170,20 +212,24 @@ public class ReceiptMarsService {
       final String assayId,
       final OtherMaterial otherMaterial,
       final MarsReceiptMessage marsMessage) {
-    final Optional<ReceiptObject> experimentReceipt = Optional.ofNullable(receipt.getExperiments())
-        .orElse(new ArrayList<>())
-        .stream()
-        .filter(s -> s.getAlias().equals(otherMaterial.id))
-        .findAny();
-    final MarsReceiptAccession marsExperimentReceipt = getExperimentMarsAccession(studyTitle, assayId, otherMaterial.id,
-        experimentReceipt);
+    final Optional<ReceiptObject> experimentReceipt =
+        Optional.ofNullable(receipt.getExperiments()).orElse(new ArrayList<>()).stream()
+            .filter(s -> s.getAlias().equals(otherMaterial.id))
+            .findAny();
+    final MarsReceiptAccession marsExperimentReceipt =
+        getExperimentMarsAccession(studyTitle, assayId, otherMaterial.id, experimentReceipt);
     if (experimentReceipt.isEmpty()) {
-      marsMessage.getErrors().add(MarsReceiptError.builder()
-          .message(
-              String.format("Cannot find an Experiment with the alias %s in the ENA receipt", otherMaterial.id))
-          .type(MarsReceiptErrorType.INVALID_METADATA)
-          .path(marsExperimentReceipt.getPath())
-          .build());
+      marsMessage
+          .getErrors()
+          .add(
+              MarsReceiptError.builder()
+                  .message(
+                      String.format(
+                          "Cannot find an Experiment with the alias %s in the ENA receipt",
+                          otherMaterial.id))
+                  .type(MarsReceiptErrorType.INVALID_METADATA)
+                  .path(marsExperimentReceipt.getPath())
+                  .build());
       return;
     }
     accessions.add(marsExperimentReceipt);
@@ -196,18 +242,23 @@ public class ReceiptMarsService {
       final String assayId,
       final DataFile dataFile,
       final MarsReceiptMessage marsMessage) {
-    final Optional<ReceiptObject> runReceipt = Optional.ofNullable(receipt.getRuns())
-        .orElse(new ArrayList<>())
-        .stream()
-        .filter(s -> s.getAlias().equals(dataFile.id))
-        .findAny();
-    final MarsReceiptAccession marsRunReceipt = getRunMarsAccession(studyTitle, assayId, dataFile.id, runReceipt);
+    final Optional<ReceiptObject> runReceipt =
+        Optional.ofNullable(receipt.getRuns()).orElse(new ArrayList<>()).stream()
+            .filter(s -> s.getAlias().equals(dataFile.id))
+            .findAny();
+    final MarsReceiptAccession marsRunReceipt =
+        getRunMarsAccession(studyTitle, assayId, dataFile.id, runReceipt);
     if (runReceipt.isEmpty()) {
-      marsMessage.getErrors().add(MarsReceiptError.builder()
-          .message(String.format("Cannot find a Run with the alias %s in the ENA receipt", dataFile.id))
-          .type(MarsReceiptErrorType.INVALID_METADATA)
-          .path(marsRunReceipt.getPath())
-          .build());
+      marsMessage
+          .getErrors()
+          .add(
+              MarsReceiptError.builder()
+                  .message(
+                      String.format(
+                          "Cannot find a Run with the alias %s in the ENA receipt", dataFile.id))
+                  .type(MarsReceiptErrorType.INVALID_METADATA)
+                  .path(marsRunReceipt.getPath())
+                  .build());
       return;
     }
     accessions.add(marsRunReceipt);
@@ -219,31 +270,36 @@ public class ReceiptMarsService {
   // ---------------------------------
 
   private MarsReceiptAccession getStudyMarsAccession(
-      final String title,
-      final Optional<ReceiptObject> studyReceipt) {
+      final String title, final Optional<ReceiptObject> studyReceipt) {
     return MarsReceiptAccession.builder()
-        .path(new MarsReceiptPath[] {
-            MarsReceiptPath.builder().key("investigation").build(),
-            MarsReceiptPath.builder().key("studies")
-                .where(MarsReceiptWhere.builder().key("title").value(title).build()).build()
-        })
+        .path(
+            new MarsReceiptPath[] {
+              MarsReceiptPath.builder().key("investigation").build(),
+              MarsReceiptPath.builder()
+                  .key("studies")
+                  .where(MarsReceiptWhere.builder().key("title").value(title).build())
+                  .build()
+            })
         .value(studyReceipt.isPresent() ? studyReceipt.get().getAccession() : null)
         .build();
   }
 
   private MarsReceiptAccession getSampleMarsAccession(
-      final String studyTitle,
-      final String sampleId,
-      final Optional<ReceiptObject> sampleReceipt) {
+      final String studyTitle, final String sampleId, final Optional<ReceiptObject> sampleReceipt) {
     return MarsReceiptAccession.builder()
-        .path(new MarsReceiptPath[] {
-            MarsReceiptPath.builder().key("investigation").build(),
-            MarsReceiptPath.builder().key("studies")
-                .where(MarsReceiptWhere.builder().key("title").value(studyTitle).build()).build(),
-            MarsReceiptPath.builder().key("materials").build(),
-            MarsReceiptPath.builder().key("samples")
-                .where(MarsReceiptWhere.builder().key("@id").value(sampleId).build()).build()
-        })
+        .path(
+            new MarsReceiptPath[] {
+              MarsReceiptPath.builder().key("investigation").build(),
+              MarsReceiptPath.builder()
+                  .key("studies")
+                  .where(MarsReceiptWhere.builder().key("title").value(studyTitle).build())
+                  .build(),
+              MarsReceiptPath.builder().key("materials").build(),
+              MarsReceiptPath.builder()
+                  .key("samples")
+                  .where(MarsReceiptWhere.builder().key("@id").value(sampleId).build())
+                  .build()
+            })
         .value(sampleReceipt.isPresent() ? sampleReceipt.get().getAccession() : null)
         .build();
   }
@@ -254,17 +310,23 @@ public class ReceiptMarsService {
       final String otherMaterialId,
       final Optional<ReceiptObject> experimentReceipt) {
     return MarsReceiptAccession.builder()
-        .path(new MarsReceiptPath[] {
-            MarsReceiptPath.builder().key("investigation").build(),
-            MarsReceiptPath.builder().key("studies")
-                .where(MarsReceiptWhere.builder().key("title").value(studyTitle).build()).build(),
-            MarsReceiptPath.builder().key("assays")
-                .where(MarsReceiptWhere.builder().key("@id").value(assayId).build()).build(),
-            MarsReceiptPath.builder().key("materials").build(),
-            MarsReceiptPath.builder().key("otherMaterials")
-                .where(MarsReceiptWhere.builder().key("@id").value(otherMaterialId).build()).build(),
-
-        })
+        .path(
+            new MarsReceiptPath[] {
+              MarsReceiptPath.builder().key("investigation").build(),
+              MarsReceiptPath.builder()
+                  .key("studies")
+                  .where(MarsReceiptWhere.builder().key("title").value(studyTitle).build())
+                  .build(),
+              MarsReceiptPath.builder()
+                  .key("assays")
+                  .where(MarsReceiptWhere.builder().key("@id").value(assayId).build())
+                  .build(),
+              MarsReceiptPath.builder().key("materials").build(),
+              MarsReceiptPath.builder()
+                  .key("otherMaterials")
+                  .where(MarsReceiptWhere.builder().key("@id").value(otherMaterialId).build())
+                  .build(),
+            })
         .value(experimentReceipt.isPresent() ? experimentReceipt.get().getAccession() : null)
         .build();
   }
@@ -275,16 +337,22 @@ public class ReceiptMarsService {
       final String dataFileId,
       final Optional<ReceiptObject> runReceipt) {
     return MarsReceiptAccession.builder()
-        .path(new MarsReceiptPath[] {
-            MarsReceiptPath.builder().key("investigation").build(),
-            MarsReceiptPath.builder().key("studies")
-                .where(MarsReceiptWhere.builder().key("title").value(studyTitle).build()).build(),
-            MarsReceiptPath.builder().key("assays")
-                .where(MarsReceiptWhere.builder().key("@id").value(assayId).build()).build(),
-            MarsReceiptPath.builder().key("dataFiles")
-                .where(MarsReceiptWhere.builder().key("@id").value(dataFileId).build()).build(),
-
-        })
+        .path(
+            new MarsReceiptPath[] {
+              MarsReceiptPath.builder().key("investigation").build(),
+              MarsReceiptPath.builder()
+                  .key("studies")
+                  .where(MarsReceiptWhere.builder().key("title").value(studyTitle).build())
+                  .build(),
+              MarsReceiptPath.builder()
+                  .key("assays")
+                  .where(MarsReceiptWhere.builder().key("@id").value(assayId).build())
+                  .build(),
+              MarsReceiptPath.builder()
+                  .key("dataFiles")
+                  .where(MarsReceiptWhere.builder().key("@id").value(dataFileId).build())
+                  .build(),
+            })
         .value(runReceipt.isPresent() ? runReceipt.get().getAccession() : null)
         .build();
   }

--- a/repository-services/isajson-ena/src/test/java/com/elixir/biohackaton/ISAToSRA/EnaReceiptToMarsTest.java
+++ b/repository-services/isajson-ena/src/test/java/com/elixir/biohackaton/ISAToSRA/EnaReceiptToMarsTest.java
@@ -1,17 +1,15 @@
-/** Elixir BioHackathon 2023 */
+/** Elixir BioHackathon 2022 */
 package com.elixir.biohackaton.ISAToSRA;
-
-import java.io.File;
-import java.nio.file.Files;
-
-import org.junit.jupiter.api.Test;
-import org.springframework.boot.test.context.SpringBootTest;
 
 import com.elixir.biohackaton.ISAToSRA.model.IsaJson;
 import com.elixir.biohackaton.ISAToSRA.sra.model.MarsReceipt;
 import com.elixir.biohackaton.ISAToSRA.sra.model.Receipt;
 import com.elixir.biohackaton.ISAToSRA.sra.service.ReceiptMarsService;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.nio.file.Files;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
 
 @SpringBootTest
 class EnaReceiptToMarsTest {

